### ISSUE TITLE
Feature request: More fine-grained control over shadow map rendering

### ISSUE THREAD

**Issue by JustASquid** [mrdoob/three.js#16901]
For my application, I need some fine-grained control over render order for a number of reasons. Thankfully the layers system offers an easy and convenient way to render specific objects in a specific order.

```js
camera.layers.set(1);
renderer.render(scene, camera);

// Render some objects with depthRead/depthWrite set differently maybe
camera.layers.set(2);
renderer.render(scene, camera);

// Perhaps apply some postprocessing effect
camera.layers.set(3);
renderer.render(scene, camera);

// Render UI after postprocessing
camera.layers.set(4);
renderer.render(scene, camera);
```

The above pattern works fine, is flexible, and (I think) is perfectly performant. However, I have found that I need more granular control over shadow map rendering. In the above example, I have an object on layer 1 that needs to display the shadow of an object on layer 3.

To fix the issue, I'd need to render *only* the shadow map with both layers 1-3 active before layer 1 is rendered 'for real'. However, it doesn't appear that there is any way to actually do this.

the function `renderer.shadowMap.render()` is exposed but calling it directly results in errors as the render state isn't set up correctly.

**Comment by JustASquid** [comment]
Friendly bump - I think that 'entire shadow map is rendered only once on first call to render()' is too restrictive and it would definitely make sense to have additional control here (In my application we currently need to disable shadows entirely!)

Alternatively, maybe the multi-stage rendering pattern shown in the first post should be discouraged in Three? What sort of alternatives can I look at?

**Comment by mrdoob** [comment]
Would a different shadow technique per light solve your issue? https://github.com/mrdoob/three.js/issues/17420

**Comment by JustASquid** [comment]
> Would a different shadow technique per light solve your issue? #17420

I don't think so. This is more about controlling where in the render pipeline the shadow map is rendered. In my use case, there is only one light.

**Comment by Mugen87** [comment]
>  I have an object on layer 1 that needs to display the shadow of an object on layer 3.

This could be solved by having layers with different semantics. However, adding an additional set of layers to control whether an object is rendered into a shadow map or not is a too specific use case to me and thus somewhat out-of-scope. 

Since the OP's requirement was not requested by other users so far, I vote to close this issue.
